scripts/spack/packages/axom/package.py

```diff
@@ -134,6 +139,10 @@ class Axom(CachedCMakePackage, CudaPackage):
     depends_on("mfem", when="+mfem")
     depends_on("mfem~mpi", when="+mfem~mpi")
 
+    # Disable fortran, causing "cannot compile a simple Fortran program"
+    # with rocm
+    depends_on("hypre~fortran", when="+mfem+rocm")
```

Comment:
Are you sure this isn't compiler specific?

Reply:
Yes, this issue seen specifically with crayftn

Reply:
hmm... I know how to check for a compiler spec in the when but I dont know how to check for a specific compiler at this point.  You also can't put an `if` around this.  This is probably going to need a feature request from Spack.

@becker33, thoughts?